Include/internal/pycore_framedata.h

```diff
@@ -0,0 +1,133 @@
+#ifndef Py_INTERNAL_FRAMEDATA_H
+#define Py_INTERNAL_FRAMEDATA_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Internal-use-only frame object constructor */
+PyFrameObject*
+_PyFrame_New_NoTrack(_Py_framedata *, int);
+
+/* These values are chosen so that the inline functions below all
+ * compare f_state to zero.
+ */
+enum _framestate {
+    FRAME_CREATED = -2,
+    FRAME_SUSPENDED = -1,
+    FRAME_EXECUTING = 0,
+    FRAME_RETURNED = 1,
+    FRAME_UNWINDING = 2,
+    FRAME_RAISED = 3,
+    FRAME_CLEARED = 4
+};
+
+typedef signed char PyFrameState;
+
+// The _Py_framedata typedef is in Include/pyframeobject.h
+struct _Py_execution_frame {
+    PyObject *globals;
+    PyObject *builtins;
+    PyObject *locals;
+    PyCodeObject *code;
+    PyFrameObject *frame_obj; // Full frame object (if created)
+    /* Borrowed reference to a generator, or NULL */
+    PyObject *generator;
+    _Py_framedata *previous;
+    int lasti;       /* Last instruction if called */
+    int stackdepth;  /* Depth of value stack */
+    int nlocalsplus;
+    PyFrameState state;       /* What state the frame is in */
+    PyObject *stack[1];
```

Comment:
The file name change obscures the changes to the field names here, where the following previously had the `f_` prefix:

* `globals`
* `builtins`
* `locals`
* `code`
* `lasti`
* `state`

To reduce the diff size, while still ensuring that the field prefix was consistently different between `PyFrameObject` and `_Py_framedata`, I went with the option of standardising the data struct on "no field prefix" (it was previously split 50/50 between that and the `f_` prefix), rather than introducing a new prefix the way I did in the original PR.

Reply:
Reverted this file rename from the PR - the internal header now remains `pycore_frame.h`, with both the full frame object and frame data struct field definitions in it.